webapp/channels/src/components/team_general_tab/team_general_tab.tsx

```diff
@@ -39,7 +37,7 @@ type State = {
     shouldFetchTeam?: boolean;
 }
 
-export default class GeneralTab extends React.PureComponent<Props, State> {
+export class GeneralTab extends React.PureComponent<Props, State> {
```

Comment:
I think the most proper typing for the props in this file would be the following

```
type Props = PropsFromRedux & OwnProps & WrappedComponentProps;
```

`WrappedComponentProps` is exported by `react-intl` and it is that, the props that the wrapped component receives from `injectIntl`.

Reply:
oh wow, this is brilliant! Indeed adding `react-intl`'s own type to Props solves all the issues. I don't need to add additional property to TeamSettings anymore. Thanks @larkox!